File: Server/src/main/java/com/basecamp/campong/domain/Post.java
package com.basecamp.campong.domain;

import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;

@Getter
@Setter
public class Post { //게시글 목록 조회시 반환할 객체
    private long postid;

    private String catename;

    private String usernick;

    private String title;

    private String location;

    private long imageid;

    private LocalDateTime uploaddate;

    private long fee;

    private String contents; // 개별 게시글 조회할 때 사용

    private String lat; // 개별 게시글 조회할 때 사용

    private String lon; // 개별 게시글 조회할 때 사용

    public Post(PostList post){
        setPostid(post.getPostid());
        setCatename(post.getCategory().getCatename());
        setUsernick(post.getUser().getUsernick());
        setTitle(post.getTitle());
        setLocation(post.getLocation());
        if(post.getItemphoto() != null)
            setImageid(post.getItemphoto().getImageid());
        setUploaddate(post.getUploaddate());
        setFee(post.getFee());
        setContents(post.getContents());
        setLat(post.getLat());
        setLon(post.getLon());
    }
}

File: Android/Campong/app/src/main/java/com/basecamp/campong/model/Post.kt
package com.basecamp.campong.model

import com.basecamp.campong.utils.Keyword
import com.google.gson.annotations.SerializedName

data class ResultPostList(
    @SerializedName("num")
    val num: Int,

    @SerializedName("data")
    val item: List<Post>,

    @SerializedName(Keyword.ERROR)
    val error: Boolean,

    @SerializedName(Keyword.ERROR_MESSAGE)
    val errInfo: String,

    @SerializedName(Keyword.ERROR_CODE)
    val errCode: Int
)

data class Post(
    @SerializedName(Keyword.POST_ID)
    val postid: Long,

    @SerializedName(Keyword.CATENAME)
    val catename: String,

    @SerializedName(Keyword.USERNICK)
    val usernick: String,

    @SerializedName(Keyword.TITLE)
    val title: String,

    @SerializedName(Keyword.LOCATION)
    val location: String,

    @SerializedName(Keyword.IMAGE_ID)
    val imageid: Long?,

    @SerializedName(Keyword.UPLOAD_DATE)
    val uploaddate: String,

    @SerializedName(Keyword.FEE)
    val fee: String
)

data class ReqUploadPost(
    @SerializedName(Keyword.CATENAME)
    val catename: String,

    @SerializedName(Keyword.TITLE)
    val title: String,

    @SerializedName(Keyword.CONTENTS)
    val contents: String,

    @SerializedName(Keyword.FEE)
    val fee: String,

    @SerializedName(Keyword.LAT)
    val lat: String,

    @SerializedName(Keyword.LON)
    val lon: String,

    @SerializedName(Keyword.LOCATION)
    val location: String,

    @SerializedName(Keyword.IMAGE_ID)
    val imageid: Long?
)

data class ResultUploadPost(
    @SerializedName(Keyword.POST_ID)
    val postid: Long,

    @SerializedName(Keyword.ERROR)
    val error: Boolean,

    @SerializedName(Keyword.ERROR_MESSAGE)
    val errInfo: String,

    @SerializedName(Keyword.ERROR_CODE)
    val errCode: Int
)

data class ReqPostView(
    @SerializedName(Keyword.POST_ID)
    val postid: Long
)

data class ResultPostView(
    @SerializedName(Keyword.POST_ID)
    val postid: Long,

    @SerializedName(Keyword.CATENAME)
    val catename: String,

    @SerializedName(Keyword.USERNICK)
    val usernick: String,

    @SerializedName(Keyword.TITLE)
    val title: String,

    @SerializedName(Keyword.LOCATION)
    val location: String,

    @SerializedName(Keyword.IMAGE_ID)
    val imageid: Long?,

    @SerializedName(Keyword.UPLOAD_DATE)
    val uploaddate: String,

    @SerializedName(Keyword.FEE)
    val fee: String,

    @SerializedName(Keyword.CONTENTS)
    val contents: String,

    @SerializedName(Keyword.ERROR)
    val error: Boolean,

    @SerializedName(Keyword.ERROR_MESSAGE)
    val errInfo: String,

    @SerializedName(Keyword.ERROR_CODE)
    val errCode: Int
)